<launch>
  <!-- only SE2 is supported -->
  <arg name="sampling_space" value="SE2" />

  <!-- load robot description -->
  <arg name="robot_model_path" value="$(find hrp5_p_description)/urdf/HRP5Pmain.urdf" />
  <param name="robot_description" command="$(find xacro)/xacro $(arg robot_model_path)" />
  <rosparam subst_value="true">
    robot/convex_path: $(find hrp5_p_description)/convex/HRP5Pmain/
  </rosparam>

  <node pkg="differentiable_rmap" type="NodeRmapSamplingFootstep" name="rmap_sampling_footstep"
        output="screen">
    <rosparam subst_value="true">
      sampling_space: $(arg sampling_space)
      support_foot_body_name: Rleg_Link5
      swing_foot_body_name: Lleg_Link5
      waist_body_name: Body
      config_path: $(find differentiable_rmap)/config/RmapSamplingHRP5PFootstep.yaml
      bag_path: /tmp/rmap_sample_set_$(arg sampling_space).bag
      sample_num: 10000
      sleep_rate: 2000
    </rosparam>
  </node>

  <node pkg="rviz" type="rviz" name="rviz"
        args="-d $(find differentiable_rmap)/rviz/RmapSamplingHRP5PFootstep.rviz"
        output="log" clear_params="true" />
</launch>
